components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsStorage.kt

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.sitepermissions
+
+import android.content.Context
+import android.support.annotation.VisibleForTesting
+import mozilla.components.feature.sitepermissions.SitePermissions.Status
+import mozilla.components.feature.sitepermissions.SitePermissions.Status.ALLOWED
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.BLUETOOTH
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.MICROPHONE
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.CAMERA
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.LOCAL_STORAGE
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.LOCATION
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.NOTIFICATION
+import mozilla.components.feature.sitepermissions.db.SitePermissionsDatabase
+import mozilla.components.feature.sitepermissions.db.toSitePermissionsEntity
+
+/**
+ * A storage implementation to save [SitePermissions].
+ *
+ */
+class SitePermissionsStorage(
+    context: Context
+) {
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var databaseInitializer = {
+        SitePermissionsDatabase.get(context)
+    }
+
+    private val database by lazy { databaseInitializer() }
+
+    /**
+     * Persists the [sitePermissions] provided as a parameter.
+     * @param sitePermissions the [sitePermissions] to be stored.
+     */
+    internal fun save(sitePermissions: SitePermissions) {
+        database
+            .sitePermissionsDao()
+            .insertOrUpdateSitePermission(
+                sitePermissions.toSitePermissionsEntity()
+            )
+    }
+
+    /**
+     * Replaces an existing SitePermissions with the values of [sitePermissions] provided as a parameter.
+     * @param sitePermissions the sitePermissions to be updated.
+     */
+    fun update(sitePermissions: SitePermissions) {
+        save(sitePermissions)
+    }
```

Comment:
Do we really need `update()` if it only calls `save()`?

Reply:
Now we have independent functions for saving and updating in `SitePermissionsDao`.